diff.py

```diff
@@ -1731,7 +1731,12 @@ def pre_process(
             splitArgs[-1] = next_row.split(".text+0x")[-1]
             args = ",".join(splitArgs)
 
-        if comment is not None and mnemonic == "bl":
+        if (
+            comment is not None
+            and next_row is not None
+            and re.search(self.config.arch.re_reloc, next_row) is None
```

Comment:
Isn't the logical thing `next_row is None or re.search(self.config.arch.re_reloc, next_row) is None`? Not that it matters in practice of course, but...

Reply:
next_row will always be set with either a relocation or the next instruction if it isn't None.

```
        next_line = lines[i] if i < len(lines) else None
        mnemonic, args = processor.pre_process(mnemonic, args, next_line, comment)
```

I check if it isn't None to validate it. If it isn't None, then it's checked with the reloc regex to determine if it has a relocation or not.

If it doesn't have a relocation in it, then continue with the mnemonic check

Reply:
Yes, I get that. But you're checking for the absence of a relocation, in which case `next_row is None` should be accepted, not rejected

Reply:
Ah, yes. You are correct that the code should continue with the next checks if `next_row` is `None` as it doesn't contain a relocation.